### PR TITLE
improve llamacpp embeddings

### PR DESCRIPTION
**Pull request by streamnsight**
- **Description:**
  Improve llamacpp embedding class by adding the `device` parameter so it can be passed to the model and used with `gpu`, `cpu` or Apple metal (`mps`).
  Improve performance by making use of the bulk client api to compute embeddings in batches.
  
  - **Dependencies:** none
  - **Tag maintainer:** 
@hwchase17

### CODE REVIEW

**Comment by streamnsight** [comment]
@hwchase17

**Comment by streamnsight** [comment]
@baskaryan 
Should be mergable now. This branch has been sitting around for too long and things changed a lot, but that was an easy fix.

**Comment by streamnsight** [comment]
@hwchase17 fixed the PR

File: libs/community/langchain_community/embeddings/llamacpp.py [hunk 1]
```diff
@@ -57,6 +57,9 @@ class LlamaCppEmbeddings(BaseModel, Embeddings):
     verbose: bool = Field(True, alias="verbose")
     """Print verbose output to stderr."""
 
+    device: Optional[str] = Field(None, alias="device")
```
Comment (baskaryan) [inline libs/community/langchain_community/embeddings/llamacpp.py:60]: should this be added for the LlamaCpp LLM as well?

**Review by ccurme — APPROVED** [review]

**Pull request merged**